alerting/src/main/kotlin/org/opensearch/alerting/util/destinationmigration/NotificationApiUtils.kt

```diff
@@ -48,6 +49,8 @@ class NotificationApiUtils {
             return try {
                 val res: GetNotificationConfigResponse = getNotificationConfig(client, GetNotificationConfigRequest(setOf(id)))
                 res.searchResult.objectList.firstOrNull()
+            } catch (e: OpenSearchSecurityException) {
+                throw e
```

Comment:
I am a little confused, what's the difference if we don't catch and throw here

Reply:
`OpenSearchSecurityException ` is a child of `OpenSearchStatusException`. It's choosing to propagate the `OpenSearchSecurityException` upwards and handle any other `OpenSearchStatusException` differently below.

Reply:
`OpenSearchSecurityException` is a subclass of `OpenSearchStatusException` and didn't want that exception to be suppressed.

Reply:
Got it, thanks.